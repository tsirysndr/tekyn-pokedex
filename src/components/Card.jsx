import React from "react";
import { View } from "react-native";

export default function Card(props) {
  return <View></View>;
}
