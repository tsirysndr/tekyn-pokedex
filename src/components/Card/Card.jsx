import React from "react";
import PropTypes from "prop-types";
import styled from "@emotion/native";

const Container = styled.View`
  height: ${(props) => props.height}px;
  width: ${(props) => props.width}px;
  background-color:  rgba(255, 255, 255, 0.8);
  border-radius: 10px;
`;

export default function Card(props) {
  return (
    <Container {...props}>
      {props.children}
    </Container>
  );
}

Card.defaultProps = {
  height: 317,
  width: 300,
  children: null,
};

Card.propTypes = {
  height: PropTypes.number,
  width: PropTypes.number,
  children: PropTypes.node,
};
