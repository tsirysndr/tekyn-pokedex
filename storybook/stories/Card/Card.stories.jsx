import React from "react";
import Card from "@Components/Card";
import { storiesOf } from "@storybook/react-native";
import CenterView from "../CenterView";

storiesOf("Card", module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add("default", () => <Card />);
