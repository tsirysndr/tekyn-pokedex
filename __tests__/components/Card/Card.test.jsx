import Card from "@Components/Card";
import { render } from "@testing-library/react-native";

test("renders the Card", () => {
  const tree = render(<Card />);
  expect(tree).toMatchSnapshot();
});
