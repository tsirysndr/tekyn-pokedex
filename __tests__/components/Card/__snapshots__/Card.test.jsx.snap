// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Card 1`] = `
<View
  height={317}
  style={
    Array [
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.8)",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
      },
      undefined,
    ]
  }
  width={300}
/>
`;
